test/unit/scan/test_folder_traversal.py

```diff
@@ -0,0 +1,195 @@
+        (tmp_path / "dir").mkdir()
+
+        (tmp_path / "dir" / "file1.txt").write_text("content1")
+
+        symlink_dir = tmp_path / "dir" / "symlink_dir"
+        symlink_dir.symlink_to(tmp_path / "dir")
+
+        folder = LocalFolder(str(tmp_path))
+
+        local_paths = folder.all_files(reporter=MagicMock())
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert absolute_paths == [
+            fix_windows_path_limit(str(tmp_path / "dir" / "file1.txt")),
+            fix_windows_path_limit(str(tmp_path / "dir" / "symlink_dir" / "file1.txt")),
+        ]
+
+    @pytest.mark.skipif(
+        platform.system() == 'Windows' and platform.python_implementation() == 'PyPy',
+        reason="Symlinks not supported on PyPy/Windows"
+    )
+    @pytest.mark.timeout(5)  # Set a 5-second timeout for this test
+    def test_folder_with_symlink_to_parent(self, tmp_path):
+
+        # Create a directory structure below with the scannig point at tmp_path/parent/child/:
+        #   tmp_path
+        #   ├── parent
+        #   │   ├── child
+        #   │   │   ├── file4.txt
+        #   │   │   └── grandchild
+        #   │   │       ├── file5.txt
+        #   │   │       └── symlink_dir -> ../../.. (symlink to tmp_path/parent)
+        #   │   └── file3.txt
+        #   ├── file1.txt
+        #   └── file2.txt
+
+        (tmp_path / "parent" / "child" / "grandchild").mkdir(parents=True)
+
+        (tmp_path / "file1.txt").write_text("content1")
+        (tmp_path / "file2.txt").write_text("content2")
+
+        (tmp_path / "parent" / "file3.txt").write_text("content3")
+
+        (tmp_path / "parent" / "child" / "file4.txt").write_text("content4")
+
+        (tmp_path / "parent" / "child" / "grandchild" / "file5.txt").write_text("content5")
+        symlink_dir = tmp_path / "parent" / "child" / "grandchild" / "symlink_dir"
+        symlink_dir.symlink_to(tmp_path / "parent")
+
+        folder = LocalFolder(str(tmp_path / "parent" / "child"))
+
+        local_paths = folder.all_files(reporter=MagicMock())
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert absolute_paths == [
+            fix_windows_path_limit(str(tmp_path / "parent" / "child" / "file4.txt")),
+            fix_windows_path_limit(str(tmp_path / "parent" / "child" / "grandchild" / "file5.txt")),
+            fix_windows_path_limit(
+                str(
+                    tmp_path / "parent" / "child" / "grandchild" / "symlink_dir" / "child" /
+                    "file4.txt"
+                )
+            ),
+            fix_windows_path_limit(
+                str(
+                    tmp_path / "parent" / "child" / "grandchild" / "symlink_dir" / "child" /
+                    "grandchild" / "file5.txt"
+                )
+            ),
+            fix_windows_path_limit(
+                str(tmp_path / "parent" / "child" / "grandchild" / "symlink_dir" / "file3.txt")
+            ),
+        ]
```

Comment:
```suggestion
        ]  # yapf: disable
```
and format the lines to be long or it's made unreadable by the line wrapper